### ISSUE TITLE
ComponentCondition doesn't work for InternetShortcut

### ISSUE THREAD

**Issue by GarrenMiller** [oleg-shilo/wixsharp#1046]
I'm in the process of building an installer using a WixSharp ManagedUI.  One of the things I am doing is optionally allowing the user to add a shortcut to the desktop and/or the Windows start menu.  The shortcut is to a web URL, so I'm using the InternetShortcut class to generate the appropriate WixEntity.  Here's the relevant piece of my `Program.cs` file (I've replaced sensitive data with obvious stand-ins, like "some.site.com"):

```
new Dir(@"%ProgramMenu%", new InternetShortcut {
    Name = $"MyApplication",
    Target = "http://some.site.com/",
    Type = InternetShortcut.ShortcutType.link,
    ComponentCondition = Condition.Create("CREATE_START_MENU_SHORTCUT=\"True\""),
    AttributesDefinition = @"IconFile=[INSTALLDIR]\app_icon.ico;IconIndex=0"
})
```

As you can see, I'm trying to create a condition via the ComponentCondition property.  This code compiles, but when I look at the resulting `.wxs` file, there is no `<Condition></Condition>` element in the `<Component>` that contains the `<InternetShortcut>`.  I decided to try the same thing with the `new File()` constructor and a dummy file, like this:

```
new File($"{ProjectPaths.InstallerRoot}dummy.txt"){
    Condition = Condition.Create("CREATE_START_MENU_SHORTCUT=\"True\"")
}
```

This worked just as I expected and created the following XML:
```
<Component Id="Component.dummy.txt_40103223" Guid="6fe30b47-2577-43ad-9095-18617f305a4d" Win64="yes">
    <Condition>CREATE_START_MENU_SHORTCUT="True"</Condition>
    <File Id="dummy.txt_40103223" Source="dummy.txt" />
</Component>
```

I was able to able to get around this issue by manually adding XML elements to the `.wxs` file, like this:
```
project.WixSourceGenerated += document => {
    var index = 0;
    document.FindAll("InternetShortcut").ForEach(element => {
        if (index < 1) {
            element.Parent.AddElement("Condition", "", "CREATE_START_MENU_SHORTCUT=\"True\"");
            index++;
        }
        else {
            element.Parent.AddElement("Condition", "", "CREATE_DESKTOP_SHORTCUT=\"True\"");
        }
    });
};
```  

I am adding two shortcuts, so hence the `forEach()`.  This works as expected, and each `<Component>` containing an `<InternetShortcut/>` now has a `<Condition>` element with my desired condition inside it.  Not a problem to do it this way, but it would be nice if I could use `ComponentCondition` instead.

**Comment by oleg-shilo** [comment]
It does look like a defect. Thank you for reporting.
Will be fixed in the next release.

**Comment by oleg-shilo** [comment]
Fixed in c27a9ce.
Will be available in the very next release.

The problem was caused by the design mistake when the `ComponentCondition` property and the user are  competing for the same backstore `AttributesDefinition` property.

The release you are using will most likely be OK if you modify the declaration like this:
```C#
var dir = new Dir(@"%ProgramMenu%", new InternetShortcut {
    Name = $"MyApplication",
    Target = "http://some.site.com/",
    Type = InternetShortcut.ShortcutType.link,
    AttributesDefinition = @"IconFile=[INSTALLDIR]\app_icon.ico;IconIndex=0"
})
. . . 
dir.ComponentCondition = Condition.Create("CREATE_START_MENU_SHORTCUT=\"True\"");
```

Otherwise, you can keep your XML-injection workaround until the next release.

**Comment by GarrenMiller** [comment]
Awesome, thank you.  I will just wait until the next release to implement this in my installer since it's not a huge issue and I would like to stay on released versions.
